app/scripts/models/auth_brokers/oauth.js

```diff
@@ -81,19 +84,32 @@ define(function (require, exports, module) {
       if (! account || ! account.get('sessionToken')) {
         return p.reject(AuthErrors.toError('INVALID_TOKEN'));
       }
-
+      let assertion;
       const relier = this.relier;
       const clientId = relier.get('clientId');
       return this._assertionLibrary.generate(account.get('sessionToken'), null, clientId)
-        .then((assertion) => {
-          var oauthParams = {
+        .then((asser) => {
+          assertion = asser;
+
+          let scopedKeyPromise = p();
```

Comment:
Since this is already inside of the `then` of a promise, it seems like lines 94-100 could be simplified:

```js
if (relier.wantsKeys()) {
  return this._provisionScopedKeys(account, assertion);
}
```